src/app/clusters/general-commissioning-server/general-commissioning-server.cpp

```diff
@@ -239,9 +294,45 @@ bool emberAfGeneralCommissioningClusterCommissioningCompleteCallback(
         }
         else
         {
+#if CHIP_CONFIG_TC_REQUIRED
+            TermsAndConditionsProvider * const termsAndConditionsProvider = Server::GetInstance().GetTermsAndConditionsProvider();
+            Optional<TermsAndConditions> acceptedTermsAndConditions;
+
+            err = termsAndConditionsProvider->GetAcceptance(acceptedTermsAndConditions);
+            if (CHIP_NO_ERROR != err)
+            {
+                //
+            }
+
+            response.errorCode =
+                CheckTermsAndConditionsAcknowledgementsState(termsAndConditionsProvider, acceptedTermsAndConditions);
+            if (CommissioningErrorEnum::kOk != response.errorCode)
+            {
+                commandObj->AddResponse(commandPath, response);
+                return true;
+            }
+
+            if (failSafe.UpdateTermsAndConditionsHasBeenInvoked())
+            {
+                // Commit terms and conditions acceptance on commissioning complete
+                err = Server::GetInstance().GetTermsAndConditionsProvider()->CommitAcceptance();
+                if (err != CHIP_NO_ERROR)
+                {
+                    ChipLogError(FailSafe, "GeneralCommissioning: Failed to commit terms and conditions: %" CHIP_ERROR_FORMAT,
+                                 err.Format());
+                    fabricTable.RevertPendingFabricData();
```

Comment:
This is too much to revert... Why is the fabric even reverted here? It should not be. This prevents error handling in commissioning.